src/com/sap/piper/k8s/ContainerMap.groovy

```diff
@@ -19,4 +20,78 @@ class ContainerMap implements Serializable {
     void setMap(Map containersMap) {
         containerMap = containersMap
     }
+
+    void initFromResource(Script script, String yamlResourceName, String buildTool) {
+        script.echo "initFromResource(yamlResourceName: $yamlResourceName, buildTool: $buildTool)"
+        Map containers = [:]
+        boolean[] piperExecutionPrepared = new boolean[1]
```

Comment:
Mh, not sure if I like this. Took me some time to understand why you are using the array with one element. 
I think it is just because you want to pass the result whether it was prepared via arguments.
Maybe there is a better approach?

Reply:
Maybe I can prepare in the outer loop.

Reply:
... not without messing up the code significantly. The problem is storing state between each iteration.

Reply:
We could have a small helper `private static class PiperExecutionPreparator` of which we pass an instance around. Tracking the state there. Will do.

Reply:
Done. Do you like it better?

Reply:
The impact on the performance is about 15 to 20 seconds longer execution to resolve every Go step always in Go only.